library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -680,9 +681,7 @@ protected void flush() throws IOException
           //until the stream is closed on the local FS, readers don't see any data.
           close();
         } else {
-          Syncable syncableOutputStream = (Syncable)outputStream;
-          syncableOutputStream.hflush();
-          syncableOutputStream.hsync();
+          ((HdfsDataOutputStream)outputStream).hsync(EnumSet.of(SyncFlag.UPDATE_LENGTH));
```

Comment:
Looking at the code closer, the files whose size isn't reflected correctly with name node are still tmp files that would be finalized later at which point their size will reflect correctly as they would be closed. It may be ok to not have the size be accurately reflected for these tmp files. How about further investigation on the complexity of using the contents of the file itself, specifically what would happen if there was a failure when writing data and there is partial data for a window.